src/rtp_sender.rs

```diff
@@ -0,0 +1,119 @@
+use anyhow::{anyhow, Result};
+use crossbeam_channel::{Receiver, Sender};
+use std::{path::PathBuf, thread};
+
+use compositor_common::{scene::Resolution, Frame};
+use compositor_pipeline::pipeline::PipelineOutput;
+use ffmpeg_next::{
+    codec::{self, Context, Id},
+    format::{self, Pixel},
+    frame, Dictionary, Rational,
+};
+
+pub struct RtpSender {
+    #[allow(dead_code)]
+    port: u16,
+    sender: Sender<Frame>,
+}
+
+impl RtpSender {
+    pub fn new(port: u16, resolution: Resolution) -> Self {
+        let port_clone = port;
+        let (sender, receiver) = crossbeam_channel::unbounded();
+        thread::spawn(move || RtpSender::run(port_clone, resolution, receiver).unwrap());
+        Self { port, sender }
+    }
+
+    fn run(port: u16, resolution: Resolution, receiver: Receiver<Frame>) -> Result<()> {
+        let mut output_ctx =
+            format::output_as(&PathBuf::from(format!("rtp://127.0.0.1:{}", port)), "rtp")?;
+        let h264_codec = codec::encoder::find(Id::H264).unwrap();
+        let mut stream = output_ctx.add_stream(h264_codec)?;
+        unsafe {
+            (*(*stream.as_mut_ptr()).codecpar).codec_id = codec::Id::H264.into();
+        }
+
+        let mut encoder = Context::new().encoder().video()?;
+        let pts_unit_secs = Rational::new(1, 90000);
+        encoder.set_time_base(pts_unit_secs);
+        encoder.set_format(Pixel::YUV420P);
+        encoder.set_width(resolution.width.try_into().unwrap());
+        encoder.set_height(resolution.height.try_into().unwrap());
+
+        let mut encoder = encoder.open_as_with(
+            h264_codec,
+            // TODO: audit settings bellow
+            // Those values are copied from somewhere, they have to be set because libx264
+            // is throwing an error if it detects default ffmpeg settings.
+            Dictionary::from_iter([
+                ("preset", "ultrafast"),
+                ("tune", "zerolatency"),
+                ("qcomp", "0.6"),
+                ("i_qfactor", "0.71"),
+                ("g", "250"),
+                ("me_range", "16"),
+                ("partitions", "+parti8x8+parti4x4+partp8x8+partb8x8"),
+            ]),
+        )?;
+        output_ctx.write_header()?;
+        let mut packet = codec::packet::Packet::empty();
```

Comment:
don't we also need to do something like `packet.rescale_ts()`?

Reply:
no, if the input timestamp has the same time_base as the output, and for video it should always be 90000.